### ISSUE TITLE
Remove some files from clients blocked by the server based on their modId/hash/file name

### ISSUE THREAD

**Issue by Shrimplonex** [Skidamek/AutoModpack#131]
Playing a fabric server, currently has JEI installed, lookign to remove it and switch to REI. I know i can force REI to install by putting it in the modpack folder, but is there any way to force it to uninstall JEI even if it's set to ignore client side mods?

**Comment by Skidamek** [comment]
what do you mean ignore client side mods?

**Comment by Shrimplonex** [comment]
I could've sworn there was a setting to have automodpack ignore the client's client side mods like iris, sodium, and not force them to get removed. But i'm wanting to get JEI to be removed from the clients. Like for instance, the server does not have Iris installed on it, however all the clients have iris and automodpack does not delete it

**Comment by Skidamek** [comment]
Do you have installed already JEI on automodpack's modpack on server?
Then look https://github.com/Skidamek/AutoModpack#do-you-want-to-delete-some-mods-from-modpack

**Comment by Shrimplonex** [comment]
Yes but removing it from the server does not make it remove from the clients as a client side mod

**Comment by Skidamek** [comment]
it does if it was previously in automodpack modpack and client had it installed by automodpack

**Comment by Shrimplonex** [comment]
it is no longer in my server, i restarted the server, i closed my client, openned myclient, logged into server, my client still has JEI isntalled, and when hovering over for instance the + on a recipe, it says it is no longer installed on the server

**Comment by Shrimplonex** [comment]
Like, the pack started off from more than just automodpack, it's a pack that had automodpack *in* it not entirely made from automodpack, so it's possible JEI was just installed manually, but that's why im asking if there's a way to force automodpack to delete it

**Comment by Shrimplonex** [comment]
![image](https://github.com/Skidamek/AutoModpack/assets/95454983/6a0b5da3-4b8f-4d9f-81ab-9a0a2dd5b614)
It also isnt pushing the update to install REI, but that's fine because i know how to force it to install a new mod, just not sure how to make it remove one

**Comment by Skidamek** [comment]
AutoModpack only removes from client mods/files removed from server's modpack. So if you want to delete some file it first needs to be installed from automodpack on client and then you can remove some file from server's modpack, regenerate it and client should receive updte which will delete deleted mod/file from server's modpack.

**Comment by Shrimplonex** [comment]
right ok so if JEI wasnt installed by automodpack, that's the poitn of my question basically, can i *force* it to remove a specific file even if automodpack didnt download it?

**Comment by Skidamek** [comment]
There is no other way yet, however i have plans to implement something like block some files by thier modId/hash/filename in the future.

**Comment by Shrimplonex** [comment]
damn alright

**Comment by Skidamek** [comment]
I think i will leave this issue as enchantment to save progress
